cdap-source-control/src/main/java/io/cdap/cdap/sourcecontrol/RepositoryManager.java

```diff
@@ -110,28 +111,30 @@ public static void validateConfig(SecureStore secureStore, SourceControlConfig s
    * doesn't return the hash for directories. If the directory contains symlinks, we can't depend on the
    * directory hash to ensure the contents are unchanged.
    *
-   * @param path The absolute path on the filesystem that should include {@link RepositoryManager#getBasePath()} as a
-   *             prefix.
+   * @param path     The absolute path on the filesystem that should include {@link RepositoryManager#getBasePath()}
+   *                 as a
+   *                 prefix.
+   * @param commitID The commit ID hash for which to get the file hash.
    * @return The git file hash of the requested file.
-   * @throws IOException              when file isn't found, there are circular symlinks or other file IO errors.
+   * @throws IOException              when file or commit isn't found, there are circular symlinks or other file IO
+   *                                  errors.
    * @throws IllegalArgumentException when the provided file isn't a regular file, eg: directory.
    * @throws NotFoundException        when the file isn't committed to Git.
-   * @throws IllegalStateException    when {@link RepositoryManager#startUp()} isn't called before this.
+   * @throws IllegalStateException    when {@link RepositoryManager#cloneRemote()} isn't called before this.
    */
-  public String getFileHash(Path path) throws IOException, NotFoundException {
+  public String getFileHash(Path path, String commitID) throws IOException, NotFoundException {
```

Comment:
Why not take a `RevCommit`? We could avoid calling parseCommit again in below code. CommitCommand returns `RevCommit` in the push flow, which we could save some steps parsing back and forth. Also I agree that perhaps taking a relativePath string is more idea since jgit works with relative path too

Reply:
We can do that, but that would mean users need to have a dependency on jgit for capturing the return type. Since operation runner is in appfabric, this would mean adding jgit to appfabric's pom.